### PR TITLE
Mark Various Functions, And Enums Public For On-Chain Price Calculations

### PR DESCRIPTION
**Pull request by bonedaddy**
# Overview

While attempting to get various bid/ask prices on-chain through CPI I ran into a few different issues due to various functions and enums not being marked as public. As far as I can tell the updated visibilities I have included in this repo are all that is needed to access this information on-chain through other programs.

### CODE REVIEW

File: dex/src/critbit.rs [hunk 1]
```diff
@@ -91,7 +91,7 @@ impl LeafNode {
     #[inline]
     pub fn price(&self) -> NonZeroU64 {
         NonZeroU64::new((self.key >> 64) as u64).unwrap()
-    }
+    }
```
Comment (armaniferrante) [inline dex/src/critbit.rs:94]:
```suggestion
    }
```

File: dex/src/critbit.rs [hunk 2]
```diff
@@ -139,7 +139,7 @@ const fn _const_max(a: usize, b: usize) -> usize {
     let gt = (a > b) as usize;
     gt * a + (1 - gt) * b
 }
-
+
```
Comment (armaniferrante) [inline dex/src/critbit.rs:142]:
```suggestion

```

File: dex/src/critbit.rs [hunk 3]
```diff
@@ -511,7 +511,7 @@ impl Slab {
                 }
                 _ => return Some(root),
             }
-        }
+        }
```
Comment (armaniferrante) [inline dex/src/critbit.rs:514]:
```suggestion
        }
```

**Comment by bonedaddy** [comment]
I've pushed another commit that marked some additional functions and the `SlabHeader` struct as public.

File: dex/src/critbit.rs [hunk 4]
```diff
@@ -252,13 +251,13 @@ const_assert_eq!(_NODE_ALIGN, align_of::<AnyNode>());
...
-    free_list_len: u64,
-    free_list_head: u32,
-
-    root_node: u32,
-    leaf_count: u64,
+pub struct SlabHeader {
+    pub bump_index: u64,
+    pub free_list_len: u64,
+    pub free_list_head: u32,
+
```
Comment (armaniferrante) [inline dex/src/critbit.rs:258]: Format :)

Comment (bonedaddy) [reply dex/src/critbit.rs]: pretty sure my most recent commit fixed the formatting issue, let me know if it doesnt.

**Comment by bonedaddy** [comment]
I have pushed [another commit](https://github.com/project-serum/serum-dex/pull/106/commits/743265c3bd22c7c42a1babfd11ae063b026a1035) with some additional changes.

**Review by armaniferrante — APPROVED** [review]

**Comment by sconybeare** [comment]
The struct fields are intentionally private in order to keep them read-only to other dex code. A better solution would be to add read-only accessor methods.

**Pull request merged**